sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/UnsafeRowUtils.scala

```diff
@@ -74,23 +92,23 @@ object UnsafeRowUtils {
       case (field, index) if UnsafeRow.isFixedLength(field.dataType) && !row.isNullAt(index) =>
         field.dataType match {
           case BooleanType =>
-            if ((row.getLong(index) >> 1) != 0L) {
+            if (getPaddingBoolean(row, index) != 0L) {
```

Comment:
I worry about creating functions in the loop `expectedSchema.fields.zipWithIndex.foreach {`. This might cause perf regression. How about to place `getPadding` and `getPaddingBoolean` in separate classes like `UnsafeRowBeUtils` and `UnsafeRowLeUtils`, and instantiate them out of the loop.

Reply:
Thanks for the review @MaxGekk! 

The `getPadding*` functions are actually created outside of the loop already. They are declared as `val` members and `UnsafeRowUtils` is a top level `object` which makes it a singleton so the `getPadding*` functions are initialized only once for the entire program; not every time they are called. They are initialized with one of the anonymous functions returned by the `if` statement. So when the `getPadding*` functions are called, only the anonymous function chosen at initialization is executed. Only a single instance of the chosen anonymous function is ever created.

It's a really nice concise short form for creating separate classes like `UnsafeRowBeUtils` and `UnsafeRowLeUtils` and choosing one of those to use ahead of time.

Reply:
Hi @MaxGekk, just wanted to follow up on this. I hope my explanation makes sense.

Reply:
Hi @MaxGekk, I just wanted to follow up with you on my comments above.

Reply:
Hi @MaxGekk, just wanted to ask if you have some time to look at my comments above

Reply:
Hi @MaxGekk, just wanted to follow up again to ask if you have some time to look at my comments above

Reply:
Hi @dongjoon-hyun, this PR has been stuck for a while. You helped me with some of my other PRs. I was wondering if you had some time to look at this one or point me to someone who can review this area of the code.

Reply:
Hi @dongjoon-hyun, @MaxGekk, just wondering if anyone has some time to look at my comments above